coul.c

```diff
@@ -752,6 +752,7 @@ void recover(void) {
             fail("unexpected log line %.3s in %s", curbuf, rpath);
     }
     fseek(rfp, 0L, SEEK_END);
+	fflush(rfp);
```

Comment:
I don't think this should be needed, there is no unflushed write - the only possible modification up to now is the ftruncate(), which is a lower-level function.

Reply:
fflush()
"and the last data modification and last file status change timestamps of the underlying file shall be marked for update." See full description: [https://pubs.opengroup.org/onlinepubs/9699919799/functions/fflush.html](https://pubs.opengroup.org/onlinepubs/9699919799/functions/fflush.html )

Reply:
The file modification time should reflect the time the file was last modified. What modification do you want to reflect here?

Note that ftruncate() also updates the modification timestamp.

Reply:
When we use fflush(rfp) (but not fflush()), we have two actions at once:
1. We clear what is possibly left in the stream buffer.
2. We give the go-ahead to change the timestamp.

Reply:
Yes, I understand what fflush does. However at this point there is nothing in the stream buffer, and there is no change that would cause us to want to change the timestamp.

That is why I asked "What modification do you want to reflect here?"